webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,254 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerturbationTestingPreview from "./PerturbationTestingPreview";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import {
+  getPertubationResponse,
+  getPertubationResponseWithoutTrainFailureRate,
+  getPertubationResponseWithFailureResponse,
+} from "mocks/api/mockPertubationAPI";
+import { formatRatioAsPercentageString } from "utils/format";
+
+const renderPerturbationTestingPreview = (
+  availableDatasetSplits: AvailableDatasetSplits
+) =>
+  renderWithRouterAndRedux(
+    <PerturbationTestingPreview
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("PerturbationTestingPreview", () => {
+  const handlers = [getPertubationResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should be having two toggle buttons for the train and test", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      expect(screen.getAllByRole("button")).toHaveLength(2);
+      expect(screen.getAllByRole("button")).not.toHaveLength(1);
+    });
+  });
+
+  it("should be having a disabled toggle button if either one of the datasplit is unavailable", async () => {
```

Comment:
Detail, but I see "datasplit" 6 times in this file, but we normally call this a "dataset split".
```suggestion
  it("should be having a disabled toggle button if either one of the dataset split is unavailable", async () => {
```